api/src/main/java/io/opentelemetry/api/trace/TraceStateBuilder.java

```diff
@@ -81,8 +86,9 @@ public TraceStateBuilder remove(String key) {
       // Copy entries from the parent.
       entries = new ArrayList<>(parent.getEntries());
     }
-    for (int i = 0; i < entries.size(); i++) {
-      if (entries.get(i).getKey().equals(key)) {
+    for (int i = 0; i < entries.size(); i += 2) {
```

Comment:
Looks like we can extract a private `doRemove` type of method that's used in both set and remove